smarts/env/wrappers/frame_stack.py

```diff
@@ -66,17 +81,29 @@ def step(
         Dict[str, bool],
         Dict[str, Dict[str, Union[float, sensors.Observation]]],
     ]:
+        """Steps the environment by one step.
+
+        Args:
+            agent_actions (Dict): Actions for each agent.
 
+        Returns:
+            Tuple[ Dict[str, List[sensors.Observation]], Dict[str, float], Dict[str, bool], Dict[str, Dict[str, Union[float, sensors.Observation]]] ]: Observation, reward, done, info, for each agent.
+        """
         env_observations, rewards, dones, infos = super(FrameStack, self).step(
             agent_actions
         )
 
         return self._get_observations(env_observations), rewards, dones, infos
 
     def reset(self) -> Dict[str, List[sensors.Observation]]:
+        """Resets the environment.
+
+        Returns:
+            Dict[str, List[sensors.Observation]]: Observation upon reset for each agent.
+        """
         env_observations = super(FrameStack, self).reset()
         for agent_id, observation in env_observations.items():
             for _ in range(self._num_stack - 1):
-                self._frames[agent_id].append(observation)
+                self._frames[agent_id].appendleft(observation)
```

Comment:
Is this for ordering observations like a queue?

Reply:
Yes, newer observation gets added to the front (left side) of the queue, instead of to the back (right side) of the queue.